--- belle-sip-1.6.3-0/src/belle_sdp_impl.c~	2016-12-06 15:34:34.000000000 +0100
+++ belle-sip-1.6.3-0/src/belle_sdp_impl.c	2018-09-26 15:32:00.385709065 +0200
@@ -1053,7 +1053,7 @@ void belle_sdp_media_description_append_
 		current_ptime=belle_sdp_mime_parameter_get_ptime(mime_parameter);
 	}
 	if (current_ptime>0){
-		char  ptime[10];
+		char  ptime[11];
 		snprintf(ptime,sizeof(ptime),"%i",current_ptime);
 		belle_sdp_media_description_set_attribute_value(media_description,"ptime",ptime);
 	}
@@ -1062,7 +1062,7 @@ void belle_sdp_media_description_append_
 		current_max_ptime=belle_sdp_mime_parameter_get_max_ptime(mime_parameter);
 	}
 	if (current_max_ptime>0){
-		char  max_ptime[10];
+		char  max_ptime[11];
 		snprintf(max_ptime,sizeof(max_ptime),"%i",current_max_ptime);
 		belle_sdp_media_description_set_attribute_value(media_description,"maxptime",max_ptime);
 	}
